.github/workflows/build-and-test.yml

```diff
@@ -75,7 +75,7 @@ jobs:
           path: /home/runner/.cache/go-build
           key: go-build-lint-${{ runner.os }}-${{ hashFiles('**/go.mod', '**/go.sum') }}
       - name: Lint
-        run: make -j4 golint
+        run: make -j2 golint
```

Comment:
Wouldn't it be better to increase the timeout itself as well?

Reply:
🤔 hmmm the timeout for golangci-lint is already set to 10m:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e52cfd022fe600d1adf85c4e90f6f0cb4482a988/.golangci.yml#L7

Reply:
I suspect that something wrong is happening after this PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7153

I see that at least `misspell` is running multiple times in every package because it validates all the nested files in each module